lib/editor/actions/map/stopStrategies.js

```diff
@@ -286,48 +283,64 @@ export function addStopToPattern (pattern: Pattern, stop: GtfsStop, index?: ?num
       if (hasShapePoints) {
         // Update shape if it exists. No need to update anything besides pattern
         // stops (which already occurred above) if there is no shape. NOTE: the
-        // behavior in this code block essentially replaces any surrounding
-        // non-stop control points with the new pattern stop and re-routes the
-        // shape between it and the surrounding stop control points.
-        // Find projected location onto pattern shape.
-        const {distanceInMeters, insertPoint} = projectStopOntoLine(stop, patternLine)
+        // behavior in this code block essentially replaces any
+        // non-stop control points in the "from" segment with the new pattern stop and re-routes the
+        // shape between it and the new stop.
+        // Non-stop control points are preserved in the new "to" segment.
+        //     ↙️ "from" segment      ↙️ "to" segment
+        //  0 ——x——— 0 ——————————x———x——— 0
+        //             ^— 0 —^ <--- new inserted stop
+
         // Add control point in order to copy of current list.
-        const controlPoint = newControlPoint(distanceInMeters, insertPoint, {
-          stopId: stop.stop_id,
-          pointType: POINT_TYPE.STOP // 2
-        })
         const stopControlPoints = controlPoints
           // TODO: refactor into shared function (see pattern-debug-lines.js)
           .map((cp, index) => ({...cp, cpIndex: index}))
           .filter(cp => cp.pointType === POINT_TYPE.STOP)
-        const {cpIndex: nextStopIndex} = stopControlPoints[index]
-        // Iterate over control points to find previous and next stop control
-        // points.
-        let spliceIndex = 1 // Init to 1 to avoid crashing in a case where we can't project the stop onto the existing pattern.
-        while (controlPoints[spliceIndex].distance < distanceInMeters && spliceIndex < nextStopIndex) {
-          spliceIndex++
-        }
         // Perform splice operation on cloned control points to remove any
         // control points between the previous and next stop control points and
         // insert new stop control point in their stead.
         const clonedControlPoints = clone(controlPoints)
         const clonedPatternSegments = clone(patternSegments)
-        // Replace n segments with 2 blank "placeholder" segments to be replaced
+
+        // Insert the new stop after the previous stopControlPoint.
+        // We assume that all control points hold for the "to" semgent (see above).
+        const previousStopControlPoint = stopControlPoints[index - 1]
```

Comment:
Do we know that index is always >1?

Reply:
We do because of https://github.com/ibi-group/datatools-ui/blob/fa19ed3de4d1399f9f1618cf7b432dd0da6d4711/lib/editor/actions/map/stopStrategies.js#L279

Reply:
But great catch, that was also what the mini-fix was addressing earlier.